modules/yugabytedb/src/main/java/org/testcontainers/containers/strategy/YugabyteDBYCQLWaitStrategy.java

```diff
@@ -0,0 +1,71 @@
+package org.testcontainers.containers.strategy;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.testcontainers.containers.Container.ExecResult;
+import org.testcontainers.containers.YugabyteDBYCQLContainer;
+import org.testcontainers.containers.wait.strategy.AbstractWaitStrategy;
+import org.testcontainers.containers.wait.strategy.WaitStrategyTarget;
+
+import static org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess;
+
+/**
+ * Custom wait strategy for YCQL API.
+ *
+ * <p>
+ * Though we can either use HTTP or PORT based wait strategy, when we create a custom
+ * keyspace/role, it gets executed asynchronously. As the wait on container.start() on a
+ * specific port wouldn't fully guarantee the custom object execution. It's better to
+ * check the DB status with this way with a smoke test query that uses the underlying
+ * custom objects and wait for the operation to complete.
+ * </p>
+ *
+ * @author srinivasa-vasu
+ */
+@RequiredArgsConstructor
+@Slf4j
+public final class YugabyteDBYCQLWaitStrategy extends AbstractWaitStrategy {
```

Comment:
this indeed now looks very convenient. I wonder if can do something similar to our existing `CassandraContainer` and rid of the cassandra driver dependency at all, at least for now.

@srinivasa-vasu to give you a little more context, on sql there are some issues due to how each database handles the scripts. We recommend using tools such as liquibase or flyway in order to execute scripts that are specialized for that kind of jobs. TBH, not sure if we will have similar issues with CQL and for that reason we will try to avoid it. IMO, we should avoid it and promote best practices around database management in general. No need to take action yet on this. @kiview WDYT?